tests/link/numba/utils.py

```diff
@@ -0,0 +1,240 @@
+    inputs,
+    assert_fn=None,
+):
+    """Function to compare python graph output and Numba compiled output for testing equality
+
+    In the tests below computational graphs are defined in Aesara. These graphs are then passed to
+    this function which then compiles the graphs in both Numba and python, runs the calculation
+    in both and checks if the results are the same
+
+    Parameters
+    ----------
+    prefix: str
+        The full string name of the module being tested.
+    fgraph: FunctionGraph
+        Aesara function Graph object
+    inputs: iter
+        Inputs for function graph
+    assert_fn: func, opt
+        Assert function used to check for equality between python and Numba. If not
+        provided uses np.testing.assert_allclose
+
+    """
+    if assert_fn is None:
+
+        def assert_fn(x, y):
+            return np.testing.assert_allclose(x, y, rtol=1e-4) and compare_shape_dtype(
+                x, y
+            )
+
+    fn_inputs = [i for i in fgraph.inputs if not isinstance(i, SharedVariable)]
+
+    aesara_py_fn = function(
+        fn_inputs, fgraph.outputs, mode=py_mode, accept_inplace=True
+    )
+    py_res = aesara_py_fn(*inputs)
+
+    aesara_numba_fn = function(
+        fn_inputs,
+        fgraph.outputs,
+        mode=numba_mode,
+        accept_inplace=True,
+    )
+    numba_res = aesara_numba_fn(*inputs)
+
+    # Get some coverage
+    eval_python_only(prefix, fn_inputs, fgraph, inputs)
+
+    if len(fgraph.outputs) > 1:
+        for j, p in zip(numba_res, py_res):
+            assert_fn(j, p)
+    else:
+        assert_fn(numba_res, py_res)
+
+    return numba_res
+
+
+def compare_rv_with_numba_output(fgraph, inputs, rng, size):
+    """Function to compare Aesara's Numba graph output and Numba output directly for
+    testing equality of RandomVariable samples.
+
+    Parameters
+    ----------
+    fgraph: FunctionGraph
+        Aesara function Graph object
+    inputs: iter
+        Inputs for function graph
+    rng: RandomState
+        RandomState variable for the current test
+    size: TensorVariable or None
+        Size of distibution to be tested
+
+    """
+    fn_inputs = [i for i in fgraph.inputs if not isinstance(i, SharedVariable)]
+
+    aesara_numba_fn = function(
+        fn_inputs,
+        fgraph.outputs,
+        mode=numba_mode,
+        accept_inplace=True,
+    )
+    numba_res = aesara_numba_fn(*inputs)
+
+    numpy_fn_name = fgraph.outputs[0].owner.op.name
+    numpy_fn = getattr(np.random, numpy_fn_name)
+
+    # This syncs the RNG states
+    numba_typify_RandomState(rng)
+
+    inputs = tuple([i.tolist() for i in inputs])
+
+    if size is None:
+
+        @numba.njit()
+        def numba_fn():
+            return numpy_fn(*inputs)
+
+    else:
+        size = tuple(size.data.astype(np.int))
+
+        @numba.njit
+        def numba_fn():
+            data = np.empty(size)
+            for i in np.ndindex(size):
+                data[i] = numpy_fn(*inputs)
```

Comment:
Just a minor question. Over here, for distributions that accept arrays as inputs. (For instance `beta`) this doesn't accept anything other than a `float` value. If so then how is it that we can pass arrays (as `dvector`s) in Aesara graphs with Numba ? (like done in `test_random.test_RandomVariable`)

Reply:
We're `numba.vectorize`ing these `np.random` functions in Aesara.  That's what makes them work with array inputs.

We can do the same for testing.